### PR TITLE
Refactor AspNetCore project

### PR DESCRIPTION
**Pull request by JanOuborny**
- Added caching type
- Structured classes in namespaces

### CODE REVIEW

File: RiotSharp.AspNetCore/Options/ApiKeyOptions.cs [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
+namespace RiotSharp.AspNetCore.Options
 {
     /// <summary>
     /// Specifies the options for ApiKey
     /// </summary>
     public class ApiKeyOptions
     {
-        internal ApiKeyOptions() { }
+        internal ApiKeyOptions() {
+            CacheType = CacheType.Internal;
```
Comment (BenFradet) [inline RiotSharp.AspNetCore/Options/ApiKeyOptions.cs]: if internal is not recommended maybe we should use another as default?

**Review by BenFradet — APPROVED** [review]
LGTM, I just have one question around the default

File: RiotSharp.AspNetCore/Options/ApiKeyOptions.cs (hunk 1 above)
Comment (Kiritsu) [reply RiotSharp.AspNetCore/Options/ApiKeyOptions.cs]: To be consistent with the rest of the lib, I'd put no cache by default.

**Comment by Kiritsu** [comment]
Shouldn't you let the user, if they want to, use their own implementation of `ICache`?

File: RiotSharp.AspNetCore/Options/ApiKeyOptions.cs (hunk 1 above)
Comment (JanOuborny) [reply RiotSharp.AspNetCore/Options/ApiKeyOptions.cs]: My bad. I wrongly thought that we use the internal implementation by default.

**Review by BenFradet — APPROVED** [review]

**Pull request merged**